doc/source/ray-core/doc_code/pattern_map_reduce.py

```diff
@@ -0,0 +1,39 @@
+import ray
+
+# __single_threaded_map_reduce_start__
+items = list(range(100))
+map_outputs = [i * 2 for i in items]
+reduce_output = sum(map_outputs)
+assert reduce_output == 9900
+# __single_threaded_map_reduce_end__
+
+
+# __parallel_map_reduce_start__
+@ray.remote
+def map(func, obj):
+    return func(obj)
+
+
+@ray.remote
+def reduce(func, *objs):
+    return func(objs)
+
+
+map_outputs = [map.remote(lambda i: i * 2, i) for i in items]
+
+# Simple reduce
+# Here we use the * syntax to unpack map_outputs
+# and pass each map output ObjectRef to reduce individually instead of passing
+# the entire list as a single argument.
+# When an argument is ObjectRef, Ray will resolve it
+# and repalce it with the actual object when calling the remote function.
```

Comment:
```suggestion
# and replace it with the actual object when calling the remote function.
```

Reply:
Can you also expand this to explain why this is desirable? Might be good to move/copy some of the text to the main doc page too.

Reply:
Discussed offline: later I will update https://docs.ray.io/en/latest/ray-core/objects.html#passing-object-arguments to include the vardiadic syntax and update the comment here to reference that.